### DISCUSSION TITLE
How to save Jax.interpreters.ad.JVPTracer as numpy array?

### DISCUSSION THREAD

**Discussion by ghost** [jax-ml/jax#12190] [Q&A]
In the latest JAX, we can print values using jax.debug.print function (https://jax.readthedocs.io/en/latest/debugging/print_breakpoint.html). However, how to save those values as a numpy array? Thank you for your time.

**Comment by sharadmv** [comment]
You can use [`jax.debug.callback`](https://jax.readthedocs.io/en/latest/_autosummary/jax.debug.callback.html) which will provide the NumPy arrays to a Python callback. You can then save them however you want.
